designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ScenarioActivitySpec.scala

```diff
@@ -0,0 +1,484 @@
+          .equalsPlainBody("The resource requires authentication, which was not supplied with the request")
+      }
+
+      "forbid access for insufficient privileges" in {
+        given()
+          .applicationState {
+            createSavedScenario(scenario, category = Category1, TestProcessingTypes.Streaming)
+          }
+          .multiPartBody(fileContent = "test", fileName = "test.xml")
+          .basicAuth("reader", "reader")
+          .when()
+          .post(s"$nuDesignerHttpAddress/api/processes/${scenarioName.value}/1/activity/attachments")
+          .Then()
+          .statusCode(403)
+          .equalsPlainBody("The supplied authentication is not authorized to access this resource")
+      }
+    }
+  }
+
+  "The scenario download attachment endpoint when" - {
+    "authenticated should" - {
+      "download existing attachment" in {
+        val attachmentId = given()
+          .applicationState {
+            createSavedScenario(scenario, category = Category1, TestProcessingTypes.Streaming)
+            createAttachment(scenarioName, fileContent)
+          }
+          .basicAuth("reader", "reader")
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/processes/${scenarioName.value}/activity")
+          .Then()
+          .extractLong("attachments[0].id")
+
+        given()
+          .basicAuth("writer", "writer")
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/processes/${scenarioName.value}/activity/attachments/$attachmentId")
+          .Then()
+          .statusCode(200)
+          .equalsPlainBody(fileContent)
+      }
+
+      "return empty body for no existing attachment" in {
+        given()
+          .applicationState {
+            createSavedScenario(scenario, category = Category1, TestProcessingTypes.Streaming)
+          }
+          .basicAuth("writer", "writer")
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/processes/${scenarioName.value}/activity/attachments/1")
+          .Then()
+          .statusCode(200)
+          .equalsPlainBody("")
+      }
+
+      "return 404 for no existing scenario" in {
+        given()
+          .basicAuth("writer", "writer")
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/processes/${wrongScenarioName.value}/activity/attachments/1")
+          .Then()
+          .statusCode(404)
+          .equalsPlainBody(s"No scenario $wrongScenarioName found")
+      }
+    }
+
+    "not authenticated should" - {
+      "forbid access for no authorization" in {
+        given()
+          .noAuth()
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/processes/${scenarioName.value}/activity/attachments/1")
+          .Then()
+          .statusCode(401)
+          .equalsPlainBody("The resource requires authentication, which was not supplied with the request")
+      }
+    }
+  }
+
+  private def createComment(scenarioName: ProcessName, commentContent: String): ValidatableResponse = {
```

Comment:
maybe it'd be better to return `Unit` instead of `ValidatableResponse`. AFAIU, we don't want to use it in the test DSL (but inside the `applicationState` part of the DSL). When the `ValidatableResponse` is returned, someone may think it can be used like that.